packages/config-eslint/moss-lint-plugin/rules/only-valid-token-names.js

```diff
@@ -0,0 +1,109 @@
+// FIXME: temporary solution with test.json file
+import themeInterface from "../test.json" with { type: "json" };
+
+const colorKeysToCssValues = (colors) => {
+  const keys = Object.keys(colors);
+  return keys.map((color) => "--color-" + color.replaceAll(".", "-"));
+};
+
+const VALID_TOKENS = colorKeysToCssValues(themeInterface.colors);
+const SELECTOR_WITH_ARBITRARY_VALUE = /(?:group-)?\w+-\[(?:var\((--[\w-]+)\)|(--[\w-]+))\]/g;
+const ARBITRARY_VALUE = /\[(?:var\((--[\w-]+)\)|(--[\w-]+))\]/;
```

Comment:
Can we use a single regex pattern with the global flag and iterate over matches using a loop? To avoid re-running regex operations 

Like: 
```typescript
const SELECTOR_WITH_ARBITRARY_VALUE =
  /(?:group-)?\w+-\[(?:var\((?<token>--[\w-]+)\)|(?<token>--[\w-]+))\]/g;
```

Reply:
replaced it with this:
```js
const ANY_TW_SELECTOR_WITH_ARBITRARY_VALUE = /\b[\w|\-:]+\[(?:var\((--[\w-]+)\)|(--[\w-]+))\]/g;
```
This pattern captures more selectors like:
- text-[--very-long-invalid-value]
- text-[var(--invalid-value)]
- hover:text-[--invalid-value]
- group-bg-[--invalid-value]
- md:background-[--primary]
- before:content-[--primary]

Instead of just capturing selectors with group or without them